Assets/MRTK/Providers/WindowsMixedReality/XR2018/WindowsMixedRealityDeviceManager.cs

```diff
@@ -933,33 +930,8 @@ private void UpdateInteractionManagerReading()
         {
             using (UpdateInteractionManagerReadingPerfMarker.Auto())
             {
-                int newSourceStateCount = InteractionManager.numSourceStates;
-                // If there isn't enough space in the cache to hold the results, we should grow it so that it can, but also
-                // grow it in a way that is unlikely to require re-allocations each time.
-                if (newSourceStateCount > interactionManagerStates.Length)
-                {
-                    interactionManagerStates = new InteractionSourceState[newSourceStateCount * InteractionManagerStatesGrowthFactor];
-                }
-
-                // Note that InteractionManager.GetCurrentReading throws when invoked when the number of
-                // source states is zero. In that case, we want to just update the number of read states to be zero.
-                if (newSourceStateCount == 0)
-                {
-                    // clean up existing controllers that didn't trigger the InteractionSourceLost event.
-                    // this can happen eg. when unity is registering cached controllers from a previous play session in the editor.
-                    // those actually don't exist in the current session and therefor won't receive the InteractionSourceLost once  
-                    // Unity's InteractionManager catches up
-                    for (int i = 0; i < numInteractionManagerStates; ++i)
-                    {
-                        RemoveController(interactionManagerStates[i].source);
-                    }
-
-                    numInteractionManagerStates = newSourceStateCount;
-                }
-                else
-                {
-                    numInteractionManagerStates = InteractionManager.GetCurrentReading(interactionManagerStates);
-                }
+                interactionManagerStates = InteractionManager.GetCurrentReading();
```

Comment:
I think that this will alloc memory on each frame, which is why we had a buffer in the first place.

It might be worth adding the comment on the previous implementation to explain why it's not using the GetCurrentReading() no param thing